homeassistant/components/tankerkoenig/__init__.py

```diff
@@ -24,11 +25,26 @@
 DEFAULT_RADIUS = 2
 DEFAULT_SCAN_INTERVAL = timedelta(minutes=30)
 
+
+def uuid4_string(value):
```

Comment:
A validator already exists for this.  Please use it instead

```
auth/providers/trusted_networks.py:                            cv.uuid4_hex,
auth/providers/trusted_networks.py:                            vol.Schema({vol.Required(CONF_GROUP): cv.uuid4_hex}),
helpers/config_validation.py:def uuid4_hex(value: Any) -> str:
```

Reply:
I already tried using this validator, but the format doesn't match. That's why I created the custom one.

The tankerkoenig API needs the UUIDs in "string" format (i.e. xxxxxx-xxxx-4xxx-xxxx-xxxxxxxx, with the dashes), but the uuid4_hex does not include the dashes.
I'd prefer not having to modify the keys delivered by tankerkoenig.de, but I'd still like to have a better validator than just "string".
I tried adding the validator generically in the PR #29321, but @balloob said I should do that in my integration instead, so I tried doing this with this PR.

Reply:
Thanks for the additional context.  You could wrap the existing validator, but its only going to save to one or two lines since you have to check the conversion.  Doesn't seem worth it.  The documentation for uuid https://docs.python.org/3/library/uuid.html doesn't do much better than `string` so this seems fine